File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int temp,mx=-1;
        for(int i = n-1 ;i>=0;i--)
           {
             temp = arr[i];
             arr[i]=mx;
             mx = max(temp,mx);
           }
        return arr;
    }
};

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        if n==1:
            arr[0]=-1
            return arr

        max = arr[-1]
        i = n-1

        while i >= 0:
            if i==n-1:
                arr[i] = -1
            else:
                 temp = arr[i]
                 arr[i]=max
                 if(temp>max):
                    max = temp
           
            i-=1

        return arr
        